.github/workflows/macOS-CMake-toolchain.yml

```diff
@@ -0,0 +1,88 @@
+name: macOS (CMake toolchain)
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: macos-latest
+
+    env:
+      # Hardcoded date selecting the latest toolchain from https://swift.org/download/#snapshots.
+      toolchain-date: 2021-01-04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set environment variables
+        run: |
+          TOOLCHAIN_DATE=${{ env.toolchain-date }}
+          # TOOLCHAIN_SHORT_DATE: TOOLCHAIN_DATE with hypens removed. Used for toolchain identifier and version.
+          TOOLCHAIN_SHORT_DATE="${TOOLCHAIN_DATE//-/}"
+          TOOLCHAIN_NAME=swift-DEVELOPMENT-SNAPSHOT-${TOOLCHAIN_DATE}-a
+          TF_TOOLCHAIN_NAME=swift-tensorflow-DEVELOPMENT-SNAPSHOT-${TOOLCHAIN_DATE}-a
+          echo "TOOLCHAIN_DATE=${TOOLCHAIN_DATE}" >> $GITHUB_ENV
+          echo "TOOLCHAIN_SHORT_DATE=${TOOLCHAIN_SHORT_DATE}" >> $GITHUB_ENV
+          echo "TOOLCHAIN_NAME=${TOOLCHAIN_NAME}" >> $GITHUB_ENV
+          echo "TF_TOOLCHAIN_NAME=${TF_TOOLCHAIN_NAME}" >> $GITHUB_ENV
+          echo "SWIFT_APIS_BUILD_DIR=/tmp/BinaryCache/tensorflow-swift-apis" >> $GITHUB_ENV
+          echo "TOOLCHAIN_ROOT_LOCATION=/Library/Developer/Toolchains/${TOOLCHAIN_NAME}.xctoolchain" >> $GITHUB_ENV
+          echo "TOOLCHAIN_HOME_LOCATION=$HOME/Library/Developer/Toolchains/${TOOLCHAIN_NAME}.xctoolchain" >> $GITHUB_ENV
+
+      - name: Install Ninja
+        run: |
+          brew install ninja
+
+      - name: Install swift.org toolchain
+        run: |
+          curl -sOL https://swift.org/builds/development/xcode/${TOOLCHAIN_NAME}/${TOOLCHAIN_NAME}-osx.pkg
+          xattr -dr com.apple.quarantine ${TOOLCHAIN_NAME}-osx.pkg
+          installer -pkg ${TOOLCHAIN_NAME}-osx.pkg -target CurrentUserHomeDirectory
+          rm -rf ${TOOLCHAIN_NAME}-osx.pkg
+
+      - name: Install pre-built TensorFlow and X10 libraries
+        run: |
+          curl -sL https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC80NTU3NC9hcnRpZmFjdE5hbWUvdGVuc29yZmxvdy1kYXJ3aW4teDY00/content?format=zip -o tensorflow-darwin-x64.zip
+          unzip tensorflow-darwin-x64.zip
+          mv tensorflow-darwin-x64/Library/tensorflow-2.4.0 ~/Library/
+
+      - name: Build tensorflow/swift-apis via CMake
+        run: |
+          set -euxo pipefail
+          mkdir -p $SWIFT_APIS_BUILD_DIR
+          cmake \
+            -B $SWIFT_APIS_BUILD_DIR \
+            -D BUILD_SHARED_LIBS=YES \
+            -D CMAKE_BUILD_TYPE=Release \
+            -D CMAKE_INSTALL_PREFIX=${TOOLCHAIN_HOME_LOCATION}/usr \
```

Comment:
If you want to inject this into the toolchain, I think that you should extract a second copy of the pkg and use that.  Something like this should do the trick:

```shell
pkg --expand ${TOOLCHAIN_NAME}-osx.pkg --destination ...
```

Reply:
I see. I wonder if installing into a separate second toolchain copy has added value? Installing into an existing single toolchain copy seems to work well.

Reply:
It does have value - it prevents the toolchain being used to build from being contaminated.  The idea is that the version which is being installed for the time being will eventually be part of the underlying image (which should be immutable).  Although the CI may have the ability to write to `/`, but imagine reproducing this where the pre-installed toolchain is in `/` and you cannot add content to it.

Reply:
Good points, thanks! Done.

(`installer -pkg <toolchain>.pkg -target <destination directory>` was the command for extracting the toolchain)